tests/unit/cli/test_cli_config_file.py

```diff
@@ -26,33 +27,41 @@ def setUp(self):
 
     def test_toml_valid_with_section(self):
         config_dir = tempfile.gettempdir()
-        configpath = Path(config_dir, "samconfig.toml")
-        configpath.write_text("version=0.1\n[config_env.topic.parameters]\nword='clarity'\n")
+        config_path = Path(config_dir, "samconfig.toml")
+        config_path.write_text("version=0.1\n[config_env.topic.parameters]\nword='clarity'\n")
         self.assertEqual(
-            TomlProvider(section=self.parameters)(config_dir, self.config_env, [self.cmd_name]), {"word": "clarity"}
+            TomlProvider(section=self.parameters)(config_path, self.config_env, [self.cmd_name]), {"word": "clarity"}
         )
 
     def test_toml_valid_with_no_version(self):
         config_dir = tempfile.gettempdir()
-        configpath = Path(config_dir, "samconfig.toml")
-        configpath.write_text("[config_env.topic.parameters]\nword='clarity'\n")
+        config_path = Path(config_dir, "samconfig.toml")
+        config_path.write_text("[config_env.topic.parameters]\nword='clarity'\n")
         with self.assertRaises(ConfigException):
-            TomlProvider(section=self.parameters)(config_dir, self.config_env, [self.cmd_name])
+            TomlProvider(section=self.parameters)(config_path, self.config_env, [self.cmd_name])
 
     def test_toml_valid_with_invalid_version(self):
         config_dir = tempfile.gettempdir()
-        configpath = Path(config_dir, "samconfig.toml")
-        configpath.write_text("version='abc'\n[config_env.topic.parameters]\nword='clarity'\n")
+        config_path = Path(config_dir, "samconfig.toml")
+        config_path.write_text("version='abc'\n[config_env.topic.parameters]\nword='clarity'\n")
         with self.assertRaises(ConfigException):
-            TomlProvider(section=self.parameters)(config_dir, self.config_env, [self.cmd_name])
+            TomlProvider(section=self.parameters)(config_path, self.config_env, [self.cmd_name])
 
     def test_toml_invalid_empty_dict(self):
         config_dir = tempfile.gettempdir()
-        configpath = Path(config_dir, "samconfig.toml")
-        configpath.write_text("[topic]\nword=clarity\n")
+        config_path = Path(config_dir, "samconfig.toml")
+        config_path.write_text("[topic]\nword=clarity\n")
 
         self.assertEqual(self.toml_provider(config_dir, self.config_env, [self.cmd_name]), {})
 
+    def test_toml_invalid_file_name(self):
+        config_dir = tempfile.gettempdir()
+        config_path = Path(config_dir, "mysamconfig.toml")
+        config_path.write_text("version=0.1\n[config_env.topic.parameters]\nword='clarity'\n")
+        config_path_invalid = Path(config_dir, "samconfig.toml")
+
+        self.assertEqual(self.toml_provider(config_path_invalid, self.config_env, [self.cmd_name]), {})
+
 
```

Comment:
Can we add a test with an absolute path? May need to have another function call generate it so it works on whatever machine runs this test, but I didn't notice that case being covered.

Reply:
I think we have a temp directory generated by `config_dir = tempfile.gettempdir()`, and that will be the absolute path.